codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java

```diff
@@ -36,8 +39,21 @@
  * Please refer to section <a href="https://tools.ietf.org/html/rfc7230#section-3.2.2">RFC 7230, 3.2.2</a>.
  */
 public class CombinedHttpHeaders extends DefaultHttpHeaders {
+    private static final Set<CharSequence> NON_COMBINABLE_HEADERS;
+
+    static {
+        NON_COMBINABLE_HEADERS = new HashSet<CharSequence>();
+
+        // According to the spec Set-cookie header does not use the list syntax and cannot be combined
+        NON_COMBINABLE_HEADERS.add(SET_COOKIE);
```

Comment:
please store this as non-mutable set

Reply:
Could we just instead special-case this one key (with a plain condition)?